diff --git a/components/esp_matter/data_model_provider/esp_matter_plugin_server_init_callbacks.cpp b/components/esp_matter/data_model_provider/esp_matter_plugin_server_init_callbacks.cpp
index 81272dc..03a41ed 100644
--- a/components/esp_matter/data_model_provider/esp_matter_plugin_server_init_callbacks.cpp
+++ b/components/esp_matter/data_model_provider/esp_matter_plugin_server_init_callbacks.cpp
@@ -69,6 +69,5 @@ void MatterWaterHeaterManagementPluginServerInitCallback() {}
 void MatterWaterHeaterModePluginServerInitCallback() {}
 void MatterCommodityTariffPluginServerInitCallback() {}
 void MatterCommodityPricePluginServerInitCallback() {}
-void MatterCommodityMeteringPluginServerInitCallback() {}
 void MatterElectricalGridConditionsPluginServerInitCallback() {}
 void MatterSoilMeasurementPluginServerInitCallback() {}
